docs/en/plugins/README.md

```diff
@@ -0,0 +1,413 @@
+angular.module('loomioApp').factory 'DiscussionTagModel', (BaseModel) ->
+  class DiscussionTagModel extends BaseModel
+    @singular: 'discussionTag'
+    @plural: 'discussionTags'
+```
+```
+# plugins/components/discussion_tags/discussion_tag_records_interface.coffee
+angular.module('loomioApp').factory 'DiscussionTagRecordsInterface', (BaseRecordsInterface, DiscussionTagModel) ->
+  class DiscussionTagRecordsInterface extends BaseRecordsInterface
+    model: DiscussionTagModel
+```
+
+*NB: These are copied straight from existing classes, with the names of the classes changed.*
+
+Then, to let our plugin know to add these classes to our asset pipeline, use the `use_asset` command:
+
+```
+plugin.use_asset 'components/discussion_tags/discussion_tag_model.coffee'
+plugin.use_asset 'components/discussion_tags/discussion_tag_records_interface.coffee'
+```
+
+`use_asset` allows us to add any type of compilable (coffee, haml, scss) file to our asset pipeline, so you can add other things this way if you find you need to as well.
+
+_NB: We could also use the `use_asset_directory` command to include a whole directory of files in one line._
+
+##### 2. Making an outlet
+
+This is the easy part. Find where in the interface you’d like your component to go, and put an ‘outlet’ directive there (if it doesn’t already exist), with a name, like so:
+
+```
+%outlet{name: “before-thread-title”}
+```
+
+You can find all of the existing outlets by searching for `%outlet` in the repository.
+
+_NB: You’ll need to pull request any new outlets against the main loomio repository, but don’t worry, we’ll definitely accept it if it’s named nicely and not right next to another outlet. We'll also be very excited to hear about what you're working on._
+
+##### 3. Making a component
+
+Now, we’ll write a little code to display tags next to the discussions title if they exist.
+
+```
+# plugins/loomio_tags/components/thread_tags/thread_tags.coffee
+angular.module('loomioApp').directive 'tags', ->
+  restrict: 'E'
+  replace: true
+  templateUrl: 'generated/components/tags.html'
+  controller: ($scope, Records) ->
+    $scope.tags = ->
+      Records.discussionTags.find(discussionId: $scope.thread.id)
+```
+
+Some things to note:
+
+- For now, plugin components must inherit the scope in which they inhabit. This means that if you put a plugin into an outlet in the ThreadPageController, then you'll have access to the ThreadPageController's $scope in your component. Here our outlet is in `thread_preview_collection.haml`, so we have access to the scope defined in `thread_preview_collection.coffee`
+- The template url format is `generated/components/<your_component_name>.html`
+- You can inject any class you'd like into the plugin controllers, which can be defined either in the core repo or within your plugin.
+- `Records` is a collection of all of our records interfaces, and `find` a method we can use to search for records within a given interface.
```

Comment:
`Records` is a collection of all of our records interfaces, and `find` **is** a method we can use to search for records within a given interface.